api/tests/DataPersister/RequestStackMockFactory.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Tests\DataPersister;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class RequestStackMockFactory {
+    public static function create(\Closure $createMock): RequestStack|MockObject {
+        $requestStack = call_user_func($createMock, RequestStack::class);
+        $request = call_user_func($createMock, Request::class);
```

Comment:
Ich hätte das jetzt als dynamischen Funktionsaufruf geschrieben weil ich das persönlich lesbarer finde als `call_user_func`:
```suggestion
        $requestStack = $createMock(RequestStack::class);
        $request = $createMock(Request::class);
```
Weisst du was der Unterschied ist? Gibts Vor- / Nachteile des einen oder anderen?

Reply:
Ich hätte jetzt das explizite $createMock->call() bevorzugt, aber da muss man ein $newThis mitgeben, was ich ja gar nicht habe in diesem kontext.
Habe es geändert. 
Ich finde es persönlich bisschen komisch, einfach hinter einer variable Klammer auf + zu zu verwenden und es so als funktion zu verwenden.
Aber wird so auch in javascript und dart gemacht.

Reply:
Es gibt sogar eine php-cs-fixer rule dafür. Ich glaube dein weg ist akzeptierter:
https://mlocati.github.io/php-cs-fixer-configurator/#version:2.18|fixer:regular_callable_call